src/common/fanctl.cpp

```diff
@@ -48,22 +49,31 @@ int8_t CFanCtlPWM::tick() {
                 pha_max = max_value - val;
                 uint8_t steps = max_value / val; // calculate number of steps
                 if (steps < 3)
-                    steps = 3;               // limit steps >= 3
-                pha_stp = max_value / steps; // calculate step - enable phase shifting
+                    steps = 3;             // limit steps >= 3
+                pha_stp = pha_max / steps; // calculate step - enable phase shifting
             } else
                 pha_stp = 0; // set step to zero - disable phase shifting
         }
 #if 1
-        else if (pha_stp) { // phase-shifting enabled
-            pha += pha_stp;
-            if (pha >= pha_max) {
-                pha_stp = -pha_stp;
-                pha = pha_max;
-            } else if (pha < 0) {
-                pha_stp = -pha_stp;
+        else
+            switch (pha_mode) {
+            case none:
                 pha = 0;
+                break;
+            case triangle:
+                pha += pha_stp;
+                if (pha >= pha_max) {
+                    pha_stp = -pha_stp;
+                    pha = pha_max;
+                } else if (pha < 0) {
+                    pha_stp = -pha_stp;
+                    pha = 0;
+                }
+                break;
+            case random:
+                pha = pha_max * ((float)rand() / RAND_MAX);
```

Comment:
the float probably won't be too healthy for the 8bit, but is ok here

Reply:
you are right, for 8bit we cannot use float, i will try to find common solution with fixpoint aritmetic or somthing similar